The patch fixes segfault when both $XDG_CONFIG_HOME and $HOME are unset
Signed-off-by: Maxim Mikityanskiy <maxtram95@gmail.com>
--- src/userpref.c.orig	2012-06-04 17:04:57.387303630 +0300
+++ src/userpref.c	2012-06-04 17:06:05.741301627 +0300
@@ -125,6 +125,9 @@
 	const char *cdir = getenv("XDG_CONFIG_HOME");
 	if (!cdir) {
 		cdir = getenv("HOME");
+		if (!cdir) {
+			cdir="/root";
+		}
 		strcpy(__config_dir, cdir);
 		strcat(__config_dir, DIR_SEP_S);
 		strcat(__config_dir, ".config");
